testing/unittests/jlibtests.cpp

```diff
@@ -3759,11 +3767,1109 @@ class PTreeSerializationDeserializationTest : public CppUnit::TestFixture
     {
         performRoundTripTest(__func__, createBinaryDataCompressionTestPTree(testXml));
     }
+
+    void testMultiThreadedSerializationAndDeserializationOfAtomTree()
+    {
+        try
+        {
+            constexpr unsigned numThreads = 50;
+            std::vector<std::thread> workers;
+            workers.reserve(numThreads);
+
+            CriticalSection errorCs;
+            bool encounteredError = false;
+            StringBuffer errorMessages;
+
+            Owned<IPropertyTree> originalTree = createPTree("MultiThreadRoot", ipt_lowmem);
+            CPPUNIT_ASSERT(originalTree != nullptr);
+
+            auto joinAll = [&workers]()
```

Comment:
This code doesn't follow the style of other similar unit tests.  E.g. see RWLockStressTest.

Since it is a unit test that doesn't matter too much, but it makes it harder to review.  (E.g. We don't tend to use condition variables - but would use a semaphore.)